app/src/main/java/com/teamagam/gimelgimel/app/network/services/GGMessageSender.java

```diff
@@ -20,20 +24,31 @@
 /**
  * Utility class handling different requests from GGMessaging server
  */
-public class GGMessagingUtils {
+public class GGMessageSender implements SharedPreferences.OnSharedPreferenceChangeListener {
 
-    private static final String LOG_TAG = GGMessagingUtils.class.getSimpleName();
+    private static final String LOG_TAG = GGMessageSender.class.getSimpleName();
 
-    /**
-     * Creates text {@link Message} with {@link MessageText} containing given text
-     * and asynchronously sends it
-     *
-     * @param message the message content text
-     */
-    public static void sendTextMessageAsync(String message) {
-        String senderId = NetworkUtil.getMac();
-        MessageText messageToSend = new MessageText(senderId, message);
-        GGMessagingUtils.sendMessageAsync(messageToSend);
+    private final GGApplication mAppContext;
+    private final PreferenceUtil mPrefs;
+    private String mSenderId;
+
+    public GGMessageSender(Context context) {
```

Comment:
Can be made static with context argument.
This way, getUsername will be usable without constructing GGMessageSender. !fix